static/app/utils/url/fetchImageData.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Fetches and provides an HTMLImageElement from a given URL.
+ * It validates the URL format before attempting to load the image.
+ *
+ * @param {string} url - The URL of the image to fetch and inspect.
+ * @returns {Promise<HTMLImageElement>}
+ * A promise that resolves to the HTMLImageElement object representing the loaded image.
+ *
+ * If the URL is invalid or the image fails to load, the promise is rejected with an error.
+ */
+export function fetchImageData(url: string): Promise<{height: number; width: number}> {
```

Comment:
Maybe I don't understand this, but is this correct return type: `Promise<{height: number; width: number}>`?

It seems to me that we never resolve object like this, but rather we resolve image object

Reply:
I would also rather return `Promise<HTMLImageElement>`

Reply:
oh yes...I updated the types and missed this one...